### PR TITLE
JS

### PR DESCRIPTION
**Pull request by juancarlospaco**

### CODE REVIEW

File: src/pylib.nim [hunk 1]
```diff
@@ -112,10 +112,11 @@ proc bool*[T](arg: T): bool =
   toBool(arg)
 
 proc input*(prompt = ""): string =
```
Comment (Yardanico) [inline src/pylib.nim:114]: Isn't it better to just hide this whole proc under the defined(js) switch as you did with others?

File: src/pylib.nim [hunk 2]
```diff
@@ -133,9 +134,10 @@ func any*[T](iter: Iterable[T]): bool =
 
 iterator items*[T](getIter: proc(): iterator(): T): T =
   ## Special items() iterator for pylib internal iterators
-  let iter = getIter()
-  while (let x = iter(); not finished(iter)):
-    yield x
+  when not defined(js):
```
Comment (Yardanico) [inline src/pylib.nim:137]: same here

File: tests/tester.nim [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+  tintdiv,
+  tclass,
+  ttonim,
+  tstring,
+  tmodulo,
+  tunpack,
+  tmisc
+
+when not defined(js):
+  include twith  # TODO: Check, is just a macro, should work?.
```
Comment (Yardanico) [inline tests/tester.nim:15]: I checked - it works fine in JS mode.

Comment (juancarlospaco) [reply tests/tester.nim:15]: No, the test do not work.

Comment (Yardanico) [reply tests/tester.nim:15]: No, I meant that the macro itself works fine, tests don't work because they rely on `open` which is IO-specific.

**Review by Yardanico — APPROVED** [review]

File: tests/tester.nim (hunk 3 above)
Comment (juancarlospaco) [reply tests/tester.nim:15]: Ye

File: src/pylib.nim [hunk 4]
```diff
@@ -260,3 +187,76 @@ template `:=`*(name, value: untyped): untyped =
   ## Mimic Pythons Operator.
   ## Creates new variable `name` and assign `value` to it.
   (var name = value; name)
+
+
+when not defined(js):
+  iterator items*[T](getIter: proc(): iterator(): T): T =
+    ## Special items() iterator for pylib internal iterators
+    let iter = getIter()
+    while (let x = iter(); not finished(iter)):
```
Comment (juancarlospaco) [inline src/pylib.nim:196]: It says `finished` is not defined for JS target (?), I dunno if compiler bug or something. :thinking: 
(proc below use this `items` too)

File: tests/tmisc.nim [hunk 5]
```diff
@@ -60,37 +60,39 @@ test "ord()":
...
-  let otherData = @[0, 1, 2, 3, 4, 5, 6, 7, 8, -1, 12412, 0, 31254, 0]
-
-  let other = filter(None, otherData)
-
-  check list(other) == @[1, 2, 3, 4, 5, 6, 7, 8, -1, 12412, 31254]
+
+when not defined(js):
+  test "bin()":
+    check bin(32587328532) == "0b11110010110010110110010110000010100"
+    check bin(-140140140140) == "-0b10000010100001000000001101011001101100"
```
Comment (juancarlospaco) [inline tests/tmisc.nim:67]: This test fails for JS target, weird JS math, I think number is too big for JS (?).

File: src/pylib.nim (hunk 4 above)
Comment (Yardanico) [reply src/pylib.nim:196]: closure iterators don't work in JS, so this code won't work

**Pull request merged**